fsspec/utils.py

```diff
@@ -482,11 +530,18 @@ def wrapper(cls):
 
 
 @contextmanager
-def nullcontext(obj):
+def nullcontext(obj: T) -> Iterator[T]:
```

Comment:
This is a generator before applying the decorator, but not a generator afterwards (it becomes a context). So, I'm not sure which state should be declared here.

Reply:
Yeah it's a little confusing. This is the correct because `contextmanager` has a type annotation that turns the iterator into a context manager for you.